### PR TITLE
fix(jsii): README.md filename is case sensitive

### LINKED ISSUE #1862: Readme.md Filename Case Sensitive for Python
### Affected Languages

- [ ] `TypeScript` or `Javascript`
- [x] `Python`
- [ ] `Java`
- [ ] .NET (`C#`, `F#`, ...)

### General Information
* **JSII Version:** 1.10.0 (build 27cfbed), typescript 3.9.7
* **Platform:**  Darwin Kernel Version 19.5.0: Tue May 26 20:41:44 PDT 2020; root:xnu-6153.121.2~2/RELEASE_X86_64 x86_64

### What is the problem?

When publishing a package to Pypi, it gets rejected with the following error message when the readme file is named like this `Readme.md`. 

```
Uploading skorfmann_cdktf_provider_aws-0.0.9-py3-none-any.whl

  0%|          | 0.00/2.73M [00:00<?, ?B/s]
  0%|          | 8.00k/2.73M [00:00<00:50, 56.0kB/s]
 10%|█         | 288k/2.73M [00:00<00:32, 79.2kB/s] 
 67%|██████▋   | 1.83M/2.73M [00:00<00:08, 113kB/s]
HTTPError: 400 Bad Request from https://upload.pypi.org/legacy/
100%|██████████| 2.73M/2.73M [00:00<00:00, 4.14MB/s]
The description failed to render for 'text/markdown'. See https://pypi.org/help/#description-content-type for more information.
Content received from server:
<html>
 <head>
  <title>400 The description failed to render for 'text/markdown'. See https://pypi.org/help/#description-content-type for more information.</title>
 </head>
 <body>
  <h1>400 The description failed to render for 'text/markdown'. See https://pypi.org/help/#description-content-type for more information.</h1>
  The server could not comply with the request since it is either malformed or otherwise incorrect.<br/><br/>
The description failed to render for &#x27;text/markdown&#x27;. See https://pypi.org/help/#description-content-type for more information.
```

I was able to fix it by changing the filename to `README.md`

It was working on my local machine - I guess due to the case insensitive filesystem on OSX. - but failing in CI.

### PR DESCRIPTION
**Pull request by RomainMuller**
The `README.md` file was processed in a case sensitive manner, but it
does not really matter what casing it uses. Changed the flow to look for
the file in a case-insensitive manner instead.

Fixes #1862

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

File: packages/jsii/lib/assembler.ts [hunk 1]
```diff
@@ -226,7 +226,12 @@ export class Assembler implements Emitter {
     }
 
     async function _loadReadme(this: Assembler) {
-      const readmePath = path.join(this.projectInfo.projectRoot, 'README.md');
+      // Search for `README.md` in a case-insensitive way
+      const fileName =
+        (await fs.readdir(this.projectInfo.projectRoot)).find(
+          (file) => file.toLocaleLowerCase() === 'readme.md',
+        ) ?? 'README.md';
+      const readmePath = path.join(this.projectInfo.projectRoot, fileName);
```
Comment (njlynch) [inline packages/jsii/lib/assembler.ts:234]: It feels a bit weird you read the files in the directory, then check to see if the path exists. I think the below reads a bit cleaner. Also, why the `await fs.readdir` vs `fs.readdirSync`?

```suggestion
      // Search for `README.md` in a case-insensitive way
      const fileName =
        fs.readdirSync(this.projectInfo.projectRoot).find(
          (file) => file.toLocaleLowerCase() === 'readme.md',
        );
      if (!fileName) { return undefined; }
      const readmePath = path.join(this.projectInfo.projectRoot, fileName);
      // Skip the fs.pathExists... check below
```

Comment (RomainMuller) [reply packages/jsii/lib/assembler.ts:234]: You're right. I guess I've been lazy 👍

File: packages/jsii/lib/assembler.ts [hunk 2]
```diff
@@ -226,7 +226,15 @@ export class Assembler implements Emitter {
...
+        (await fs.readdir(this.projectInfo.projectRoot)).find(
+          (file) => file.toLocaleLowerCase() === 'readme.md',
+        );
+      if (fileName == null) {
+        return undefined;
+      }
+      const readmePath = path.join(this.projectInfo.projectRoot, fileName);
       if (!(await fs.pathExists(readmePath))) {
         return undefined;
       }
```
Comment (njlynch) [inline packages/jsii/lib/assembler.ts:240]: Are these lines still needed? Trying to think of a scenario where `readdir` matches a file but `pathExists` returns false.

Comment (RomainMuller) [reply packages/jsii/lib/assembler.ts]: 🤦🏻‍♂️

**Pull request merged**